### PR TITLE
[editorial] Inline Hugo front matter

### LINKED ISSUE #4029: [editorial] Inline Hugo front matter
- Context: https://github.com/open-telemetry/opentelemetry-specification/pull/4021#issuecomment-2088745587
- The proposal is to inline the Hugo front matter as actual YAML front matter rather than via an HTML comment. This will streamline publication to the OTel website, and avoid issues like #4021

/cc @reyang @carlosalberto @open-telemetry/docs-maintainers

### PR DESCRIPTION
**Pull request by chalin**
- Fixes #4029
- Replacement solution for #4021
- Notes & questions:
  - I kept "Hugo front matter used to generate the website version of this page" as a YAML comment. Shall I drop those lines now that it's official Markdown front matter?
  - Because markdownlint actually recognizes front-matter titles, I've disabled that feature so that it doesn't report an error (due to the title also appearing in the page)

/cc @reyang @carlosalberto @svrnm

### CODE REVIEW

File: specification/configuration/file-configuration.md [hunk 1]
```diff
@@ -387,7 +388,7 @@ try {
     // Create SDK components from configuration model
     openTelemetry = FileConfiguration.create(configurationModel);
 } catch (Throwable e) {
-    log.error("Error initializing SDK from configuration file", e);    
+    log.error("Error initializing SDK from configuration file", e);
```
Comment (reyang) [inline specification/configuration/file-configuration.md:391]: Good catch!

**Review by reyang — APPROVED** [review]

**Comment by chalin** [comment]
I'm looking into the `markdown-toc` GH action failure.

**Comment by chalin** [comment]
Hmm, the `markdown-toc` tool seems to be cleaning up more than just the TOC. It strips the comment from the front matter and removes any blank lines between the title and the front matter 🤷🏼‍♂️. I'll commit the change so that you can see.

**Comment by chalin** [comment]
So strange, `markdown-toc` is also adding blank lines at the end of files!

**Comment by chalin** [comment]
Ok, well ... due to unresolved bugs in `markdown-toc`, we can inline Hugo front matter 🤷🏼‍♂️
Ref.: https://github.com/jonschlinkert/markdown-toc/issues/151#issuecomment-2105313699.

**Comment by chalin** [comment]
I've removed the commits that contained the `markdown-toc` mangled content. So there should be only one GH action failure, the `markdown-toc` check.

So, it seems that this can't be merged until `markdown-toc` is fixed. :-/

**Comment by chalin** [comment]
I could modify the `markdown-toc` make rule to temporarily comment out the front matter, but I'm not sure that it's worth it. WDYT @reyang et all?

**Review by jsuereth — APPROVED** [review]

**Comment by jsuereth** [comment]
Approved pending TOC fix.

**Review by arminru — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by chalin** [comment]
Ok, I'll tackle on the `markdown-toc` workaround as soon as I can.

**Comment by chalin** [comment]
Nope, not stale, I was OOO.

**Comment by chalin** [comment]
Heads up that I'm really swamped atm, so I'm unsure when I'll be able to get to this.

**Comment by chalin** [comment]
This is still relevant and I'll get to it as soon as I can.

**Comment by chalin** [comment]
Just curious if you've considered use of https://www.npmjs.com/package/doctoc instead? I'll investigate this as a possible less buggy alternative, and/or propose a workaround.

**Comment by chalin** [comment]
Still valid. Will get to it eventually 😄.

**Comment by svrnm** [comment]
let's keep this open for now, until @chalin gets to it

**Comment by chalin** [comment]
Still relevant.

**Comment by chalin** [comment]
Still relevant, though I don't know when I'll get time to work on it. Can we tell the bot to stop marking it as stale? Btw, I think that resolving https://github.com/open-telemetry/semantic-conventions/issues/1009 across spec repos is at least as important as this issue here.

**Comment by chalin** [comment]
Still backloged; still relevatn.

**Comment by chalin** [comment]
I don't know if converting to draft will help appease our vigilant sentinel bot. I appreciate the reminders, and will get to it eventually (closer to EOY probably) -- though I'll likely have to restart from scratch. At this point, I'm ok with closing or keeping this open until I do get to work on it.

**Comment by chalin** [comment]
Closing. I've added a reference to this PR and the corresponding issue it resolve to my personal issue tracker. I'll get back to it as soon as I can. If you feel that the urgency of working on this has increased, please add a comment to that effect on:

- #4029

**Pull request closed without merging**